internal/k8s/cache/gateway.go

```diff
@@ -0,0 +1,75 @@
+// Copyright Project Contour Authors
```

Comment:
I think I'd put these into an `internal/reconciler` or `internal/controller` package, I think that reflects what they do better.